packages/common/src/grpc-operation.js

```diff
@@ -0,0 +1,206 @@
+/*!
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module common/grpcOperation
+ */
+
+'use strict';
+
+var events = require('events');
+var modelo = require('modelo');
+
+/**
+ * @type {module:common/grpcService}
+ * @private
+ */
+var GrpcService = require('./grpc-service.js');
+
+/**
+ * @type {module:common/grpcServiceObject}
+ * @private
+ */
+var GrpcServiceObject = require('./grpc-service-object.js');
+
+/**
+ * An Operation object allows you to interact with APIs that take longer to
+ * process things.
+ *
+ * @constructor
+ * @alias module:common/grpcOperation
+ *
+ * @param {module:common/grpcService|module:common/grpcServiceObject} parent - The
+ *     parent object. This should be configured to use the longrunning.operation
+ *     service.
+ * @param {string} name - The operation name.
+ */
+function GrpcOperation(parent, name) {
```

Comment:
Probably. The current implementation of `module:speech/operation` is slightly different in that when the metadata is retrieved it is mixed into the `operation` instance on which you called `.on('complete, ...)`, and then a reference to the `operation` is passed to the callback, e.g.

```js
// In module:speech/operation
operation.on('complete', function (_operation, apiResponse) {
  operation === _operation; // true
});
```

whereas `GrpcOperation` just does:

```js
operation.on('complete, function (apiResponse) {
  operation === apiResponse; // false
});
```